### DISCUSSION TITLE
"ERROR:cflib.crazyflie.mem.deck_memory:Version 3 not supported"

### DISCUSSION THREAD

**Discussion by crazyflie-rugarma** [bitcraze/discussions#947] [Q&A]
Dear bitcraze support community,

I am attempting to flash the Wi-Fi example onto my Crazyflie, but it appears to be encountering an incompatibility issue when I follow the steps outlined in the [tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/). I am encountering the following error during the flashing of the Wi-Fi image streamer actions:

![Version3_not_supported](https://github.com/bitcraze/discussions/assets/146962094/d87a661c-af58-4b96-bc4b-8c6276a3de5e)

`cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w radio://0/80/2M/E7E7E7E7E7`

Reset to bootloader mode ... --> After this line, the drone emits four beeps, as is customary when it is being powered on.
Could not save cache, no writable directory
Could not save cache, no writable directory --> Then bips 4 times again
Version 3 not supported_

I have located [this](https://github.com/bitcraze/crazyflie-lib-python/blob/master/cflib/crazyflie/mem/deck_memory.py) portion of the code where it appears to announce this compatibility issue.

![version_3_not_supported_line](https://github.com/bitcraze/discussions/assets/146962094/82e8a871-eb59-4897-846b-695ac5dded7f)

The Crazyflie displays various LED patterns, which I will describe just in case it provides any clues about the problem.

- There are two green LEDs, with one appearing to blink at double the speed of the other.
- One red LED blinks as rapidly as the fastest green LED.
- Additionally, there are two static blue LEDs.

It has only one deck attached to crazyflie: AI deck 1.1. Rev C1

Regarding the Crazyflie client log, among other things, it has detected the presence of one deck, which I believe to be the AI Deck (as it is the only one connected to the Crazyflie). I have not found this line " ESP32: I (910) SYS: Initialized." specified in [web](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example) but I have tried to program firmware (2023.07 - firmware-cf2-2023.07.zip) several times
![health_status](https://github.com/bitcraze/discussions/assets/146962094/68dbfa33-65dd-4399-87e3-be883d118b16)

In order to find a solution on my own, I attempted to flash the bootloader on the GAP8 separately using an Olimex ARM-USB-TINY-H, as a precaution to resolve the issue, but unfortunately, I was not successful. I encountered an error message: 'Error: burst read failed.' According to the [guide](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/), this error suggests that 
"_NOTE: if you see Error: Burst read failed or something similar, make sure that the 10-pin lint cable is properly attached. The programmer is not able to connect to the AI deck properly_". 

I could not check if it was the solution, because I did not made it.

The connections are being made as follows:

- The PC USB port detects the Olimex within the VM.
- The Olimex is connected to the ARM-JTAG-SWD.
- The ARM-JTAG-SWD is connected via a set of male-male "dupont wires" to the ARM-JTAG-20-10.
- The ARM-JTAG-20-10 is connected to the drone's "GAP8 JTAG connector" in the correct position as indicated by the [guide](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/flashing/) and [datasheet](https://www.bitcraze.io/documentation/hardware/ai_deck_1_1/ai_deck_1_1-datasheet.pdf).

Could you please provide some guidance?

Thank you very much in advance, and best regards

**Comment by krichardsson** [comment]
This looks like a compatibility problem between the python lib and Crazyflie firmware.

It should work if you upgrade the python lib to the latest version and flash the Crazyflie (with the AI-deck mounted) through the client with the latest release.
